proto/extension/service.proto

```diff
@@ -0,0 +1,93 @@
+    rpc PostVirtualHostModify(PostVirtualHostModifyRequest) returns (PostVirtualHostModifyResponse) {};
+
+    // PostHTTPListenerModify allows an extension to modify Listeners before they are finalized
+    // PostHTTPListenerModify will always be executed after each Listener is generated by Envoy Gateway
+    // so long as the extension has provided config in the EnvoyGateway resource to opt-into this Hook
+    rpc PostHTTPListenerModify(PostHTTPListenerModifyRequest) returns (PostHTTPListenerModifyResponse) {};
+
+    // PostTranslateInsert allows an extension to inject additional clusters and/or secrets to the xDS config
+    // PostTranslateInsert will always be executed after Envoy Gateway has generated an xDS config
+    // so long as the extension has provided config in the EnvoyGateway resource to opt-into this Hook
+    rpc PostTranslateInsert(PostTranslationInsertRequest) returns (PostTranslationInsertResponse) {};
+}
+
+// PostRouteModifyRequest sends a Route that was generated by Envoy Gateway along with context information to an extension so that the Route can be modified
+message PostRouteModifyRequest {
+    envoy.config.route.v3.Route route = 1;
+    PostRouteExtensionContext route_context = 2;
+}
+
+
+// PostRouteModifyResponse is the expected response from an extension and contains a modified version of the Route that was sent
+// An extension may choose to return a nil Route to cause Envoy Gateway to discard it
+message PostRouteModifyResponse {
+    envoy.config.route.v3.Route route = 1;
+}
+
+
+// PostVirtualHostModifyRequest sends a VirtualHost that was generated by Envoy Gateway along with context information to an extension so that the VirtualHost can be modified
+message PostVirtualHostModifyRequest {
+    envoy.config.route.v3.VirtualHost virtual_host = 1;
+    PostVirtualHostExtensionContext virtual_host_context = 2;
+}
+
+// PostVirtualHostModifyResponse is the expected response from an extension and contains a modified version of the VirtualHost that was sent
+// An extension may choose to return a nil VirtualHost to cause Envoy Gateway to discard it
+message PostVirtualHostModifyResponse {
+    envoy.config.route.v3.VirtualHost virtual_host = 1;
+}
+
+// PostVirtualHostModifyRequest sends a Listener that was generated by Envoy Gateway along with context information to an extension so that the Listener can be modified
+message PostHTTPListenerModifyRequest {
+    envoy.config.listener.v3.Listener listener = 1;
+    PostHTTPListenerExtensionContext listener_context = 2;
+}
+
+
+// PostHTTPListenerModifyResponse is the expected response from an extension and contains a modified version of the Listener that was sent
+// An extension may choose to return a nil Listener to cause Envoy Gateway to discard it
+message PostHTTPListenerModifyResponse {
+    envoy.config.listener.v3.Listener listener = 1;
+}
+
+
+// PostTranslationInsertRequest currently sends only context to an extension
```

Comment:
can we only have modify ?

Reply:
was recommending to use `modify` so the end returned result can be an addition or deletion of a list of resources